File: MATEMÁTICA/2514/main.c
#include <stdio.h>

int gcd(int a, int b) {
    if (b == 0) return a;
    return gcd(b, a % b);
}

int lcm(int a, int b) {
    return a * b / gcd(a, b);
}

int nextAlignment(int m, int l1, int l2, int l3) {
    return lcm(lcm(l1, l2), l3) - m;
}

int main() {
    int m, l1, l2, l3;
    while (scanf("%d", &m) != EOF) {
        scanf("%d %d %d", &l1, &l2, &l3);
        printf("%d\n", nextAlignment(m, l1, l2, l3));
    }
    return 0;
}

File: MATEMÁTICA/2514/main.rb
def gcd(a, b)
  return a if b.zero?
  gcd(b, a % b)
end

def lcm(a, b)
  a * b / gcd(a, b)
end

def next_alignment(m, l1, l2, l3)
  lcm(lcm(l1, l2), l3) - m
end

while (m = gets&.to_i)
  l1, l2, l3 = gets.split.map(&:to_i)
  puts next_alignment(m, l1, l2, l3)
end
